src/text_file_functions.cpp

```diff
@@ -905,15 +904,31 @@ QString AOApplication::get_effect(QString effect, QString p_char,
 QString AOApplication::get_effect_property(QString fx_name, QString p_char,
                                            QString p_property)
 {
-  QString f_property;
-  if (p_property == "sound")
-    f_property = fx_name;
-  else
-    f_property = fx_name + "_" + p_property;
-
-  QString f_result = get_config_value(f_property, "effects.ini", current_theme, get_subtheme(), default_theme, read_char_ini(p_char, "effects", "Options"));
-  if (f_result == "")
-      f_result = get_config_value(f_property, "effects/effects.ini", current_theme, get_subtheme(), default_theme, "");
+  const auto paths = get_asset_paths("effects/effects.ini", current_theme, get_subtheme(), default_theme, "");
```

Comment:
Unless I am misreading this we are running this (complex) chunk of code anytime an IC message is received.
That is an utter waste of runtime if we are being honest here.

Maybe cache the results of your lookup instead of looking it up new each time we need an effect value.